ide/che-core-ide-stacks/src/main/resources/stacks.json

```diff
@@ -2771,5 +2771,50 @@
     "name": "type-blank.svg",
     "mediaType": "image/svg+xml"
   }
- }
+  },
+  {
+    "id": "openshift",
+    "name": "openshift",
+    "description": "Openshift Workspace",
+    "scope": "general",
+    "tags": [
+      "Openshift"
+    ],
+    "workspaceConfig": {
+      "environments": {
+        "default": {
+          "machines": {
+            "dev-machine/main": {
+              "installers": [
+                "org.eclipse.che.exec", "org.eclipse.che.terminal", "org.eclipse.che.ws-agent"
+              ],
+              "servers": {},
+              "attributes" : {
+                "memoryLimitBytes": "2147483648"
+              }
+            }
+          },
+          "recipe": {
```

Comment:
@benoitf As it is an initial implementation it doesn't have to cover all the use cases. Implementing k8s here seems more interesting since it was quite impossible in the master.
And points 2 and 3 (GA, yes, it's by design) from Gennady's comment would allow using env types from the master in Openshift infra.